apps/explorer/src/app/components/txs/tx-filter.tsx

```diff
@@ -0,0 +1,181 @@
+import { t } from '@vegaprotocol/i18n';
+import {
+  DropdownMenu,
+  DropdownMenuCheckboxItem,
+  DropdownMenuContent,
+  DropdownMenuItemIndicator,
+  DropdownMenuSeparator,
+  DropdownMenuSub,
+  DropdownMenuSubTrigger,
+  DropdownMenuTrigger,
+  Icon,
+  DropdownMenuSubContent,
+} from '@vegaprotocol/ui-toolkit';
+import classNames from 'classnames';
+import type { Dispatch, SetStateAction } from 'react';
+
+// All possible transaction types. Should be generated.
+export type FilterOption =
+  | 'Amend LiquidityProvision Order'
+  | 'Amend Order'
+  | 'Batch Market Instructions'
+  | 'Cancel LiquidityProvision Order'
+  | 'Cancel Order'
+  | 'Cancel Transfer Funds'
+  | 'Chain Event'
+  | 'Delegate'
+  | 'Ethereum Key Rotate Submission'
+  | 'Issue Signatures'
+  | 'Key Rotate Submission'
+  | 'Liquidity Provision Order'
+  | 'Node Signature'
+  | 'Node Vote'
+  | 'Proposal'
+  | 'Protocol Upgrade'
+  | 'Register new Node'
+  | 'State Variable Proposal'
+  | 'Submit Oracle Data'
+  | 'Submit Order'
+  | 'Transfer Funds'
+  | 'Undelegate'
+  | 'Validator Heartbeat'
+  | 'Vote on Proposal'
+  | 'Withdraw';
+
+// Alphabetised list of transaction types to appear at the top level
+export const PrimaryFilterOptions: FilterOption[] = [
+  'Amend LiquidityProvision Order',
+  'Amend Order',
+  'Batch Market Instructions',
+  'Cancel LiquidityProvision Order',
+  'Cancel Order',
+  'Cancel Transfer Funds',
+  'Delegate',
+  'Liquidity Provision Order',
+  'Proposal',
+  'Submit Oracle Data',
+  'Submit Order',
+  'Transfer Funds',
+  'Undelegate',
+  'Vote on Proposal',
+  'Withdraw',
+];
+
+// Alphabetised list of transaction types to nest under a 'More...' submenu
+export const SecondaryFilterOptions: FilterOption[] = [
+  'Chain Event',
+  'Ethereum Key Rotate Submission',
+  'Issue Signatures',
+  'Key Rotate Submission',
+  'Node Signature',
+  'Node Vote',
+  'Protocol Upgrade',
+  'Register new Node',
+  'State Variable Proposal',
+  'Validator Heartbeat',
+];
+
+export const AllFilterOptions: FilterOption[] = [
+  ...PrimaryFilterOptions,
+  ...SecondaryFilterOptions,
+];
+
+export function getFilterLabel(filters: Set<string>) {
+  if (!filters || filters.size !== 1) {
+    return <span>{t('Filter')}</span>;
+  }
+
+  return (
+    <span>
+      {t('Filters')}: <code>{Array.from(filters)[0]}</code>
+    </span>
+  );
+}
+
+// Effectively a copy & paste of the Vega DropdownMenuItem styles
+const itemClass = classNames(
```

Comment:
Suggestion (non-blocking): as above, it'd be nice maintainability win to import exported styles from the dropdownMenuItem element

Reply:
As above. Fixed.